File: test/fixtures/clojure/put_xput.clj
(require '[cheshire.core :as json])
(require '[clj-http.client :as client])

(client/put "http://localhost:28139/twitter/_mapping/user?pretty" {:form-params {:properties {:email {:type "keyword"}}}
                                                                   ;; :body "{\"properties\": {\"email\": {\"type\": \"keyword\"}}}"
                                                                   :content-type :json})

File: test/fixtures/kotlin/put_xput.kt
import java.io.IOException
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody

val client = OkHttpClient()

val MEDIA_TYPE = "application/json".toMediaType()

val requestBody = "{\"properties\": {\"email\": {\"type\": \"keyword\"}}}"

val request = Request.Builder()
  .url("http://localhost:28139/twitter/_mapping/user?pretty")
  .put(requestBody.toRequestBody(MEDIA_TYPE))
  .header("Content-Type", "application/json")
  .build()

client.newCall(request).execute().use { response ->
  if (!response.isSuccessful) throw IOException("Unexpected code $response")
  response.body!!.string()
}
